### PR TITLE
Refactors Shield Generators to use Beam(), visual improvement

### PR DESCRIPTION
**Pull request by Iamgoofball**
dreamseeker_2016-05-23_18-01-25

### CODE REVIEW

File: code/modules/power/singularity/field_generator.dm [hunk 1]
```diff
@@ -288,15 +295,19 @@ field_generator power level display
 			G.fields += CF
 			for(var/mob/living/L in T)
 				CF.Crossed(L)
-
+	*/
+	var/datum/beam/CF = Beam(BeamTarget = G, icon_state="Contain_F",icon='icons/obj/singularity.dmi',time=72000, maxdistance=10,beam_type=/obj/effect/ebeam/containment)
```
Comment (ChangelingRain) [inline code/modules/power/singularity/field_generator.dm]: 2 hours is a long time, but I'd prefer it actually being infinite.

Comment (Iamgoofball) [inline code/modules/power/singularity/field_generator.dm]: tried -1 and it didnt work

**Comment by PKPenguin321** [comment]
so wait can they go diagonal now and shit?

**Comment by Iamgoofball** [comment]
in theory yea, we'd need to alter the actual connecting code tho

**Comment by ike709** [comment]
diagonal containment atleast in a later PR pls

**Comment by Dorsisdwarf** [comment]
I cant wait for the bug reports of "SINGULO LOOSE" any time the shit hits 120 minutes.

File: code/modules/power/singularity/field_generator.dm (hunk 1 above)
Comment (JJRcop) [inline code/modules/power/singularity/field_generator.dm]:
```DM
time=INFINITY,
```

File: code/modules/power/singularity/containment_field.dm [hunk 2]
```diff
@@ -98,16 +58,18 @@
 		bump_field(user)
 	return
 
-/obj/machinery/field/proc/clear_shock()
-	hasShocked = 0
-
-/obj/machinery/field/proc/bump_field(atom/movable/AM as mob|obj)
-	if(hasShocked)
-		return 0
-	hasShocked = 1
```
Comment (phil235) [inline code/modules/power/singularity/containment_field.dm]: if you remove hasShocked you're going to get an infinite loop of bump_field() when a mob is stuck between a containment field and a wall, getting thrown repeatedly. We had that var specifically to stop this. Look at the history of the file.

Comment (xxalpha) [inline code/modules/power/singularity/containment_field.dm]: Maybe we should preserve these objects but make them invisible.

**Comment by Cheridan** [comment]
@Iamgoofball does this fix #15704 ?

**Comment by RemieRichards** [comment]
@Cheridan yes it should do, since I made non moving Beam()s skip most of the update code, and since these beams can only exist statically, it shouldn't ever have to redraw (bar full addition and full removal)
tl;dr yeah most likely.

**Comment by optimumtact** [comment]
Merge conflict and needs the feedback adressed from phil and Joan/JJRCop

**Comment by KorPhaeron** [comment]
goof let me know if you're going to finish this and I"ll reopen it

**Comment by Iamgoofball** [comment]
It just needed a merge conflict fix

**Comment by optimumtact** [comment]
you need to address phils comment as well

**Comment by Iamgoofball** [comment]
yeah good point
